--- a/screen.c	2016-02-05 19:59:25.000000000 +0000
+++ b/screen.c	2016-02-05 20:00:45.000000000 +0000
@@ -113,6 +113,9 @@
 FILE *dfp;
 #endif
 
+#ifdef _AIX
+# include <sys/socket.h> /* AF_INET6 */
+#endif
 
 extern char Term[], screenterm[], **environ, Termcap[];
 int force_vt = 1;
--- a/os.h	2016-02-05 20:53:09.000000000 +0000
+++ b/os.h	2016-02-05 20:53:48.000000000 +0000
@@ -127,8 +127,10 @@
 #endif
 
 #ifdef SYSV
+#ifndef _AIX
 # define index strchr
 # define rindex strrchr
+#endif
 # define bzero(poi,len) memset(poi,0,len)
 # define bcmp memcmp
 # define killpg(pgrp,sig) kill( -(pgrp), sig)
--- a/tty.sh	2016-02-05 20:55:40.000000000 +0000
+++ b/tty.sh	2016-02-05 20:56:42.000000000 +0000
@@ -1508,8 +1508,9 @@
   struct stat st;
   char * real;
   int rc;
+  char *resolved_name = malloc(PATH_MAX +1);
 
-  real = realpath(tty, NULL);
+  real = realpath(tty, resolved_name);
   if (!real)
     return -1;
 
--- a/utmp.c	2016-02-05 20:54:24.000000000 +0000
+++ b/utmp.c	2016-02-05 20:55:13.000000000 +0000
@@ -631,7 +631,11 @@
   /* must use temp variable because of NetBSD/sparc64, where
    * ut_xtime is long(64) but time_t is int(32) */
   (void)time(&now);
+#ifdef _AIX
+  u->ut_time = now;
+# else
   u->ut_tv.tv_sec = now;
+#endif
 }
 
 static slot_t
